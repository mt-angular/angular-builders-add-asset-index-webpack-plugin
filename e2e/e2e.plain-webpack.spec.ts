import { addAssetIndexPluginList } from './webpack-config/add-asset-index-plugin-configurations';
import path from 'path';
import { execSyncCommand, readFileAsync } from './util';



const rootProject = path.resolve(__dirname, '..');

const directories = {
    webpackConfig: path.join(__dirname, 'webpack-config'),
    root: rootProject,
    dist: path.join(__dirname, 'dist')
};



const commands = {
    compileProject: `npx tsc --project ${rootProject}`,
    compileWebpackConfigs: `npx tsc --project ${directories.webpackConfig}`,
    runWebpack: mode => `npx webpack --mode ${mode} --config ${path.join(directories.webpackConfig, 'webpack.config.js')}`
};


console.log('Compile Project', commands.compileProject);
execSyncCommand(commands.compileProject);

console.log('Compile Webpack', commands.compileWebpackConfigs);
execSyncCommand(commands.compileWebpackConfigs);



const modes: Array<'development' | 'production'> = [ 'development', 'production' ];
const configs: {
    development: Array<string>[]
    production: Array<string>[]
} = {} as any;

for (const mode of modes) {
    console.log('Execute Webpack', commands.runWebpack(mode));
    execSyncCommand(commands.runWebpack(mode));

    const webpackConfigurations = addAssetIndexPluginList({ mode });
    configs[ mode ] = webpackConfigurations.map(config => [ config.title, config.outputDir ]);
}


describe.each(modes)(
    'Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = %s End2End',
    (mode: 'development' | 'production') => {


        /* beforeAll(() => {
             jest.setTimeout(30000);
        }); */
        test.each(configs[ mode ])(
            'Snapshot index.html: %s', async (title: string, outputDir: string) => {
                const indexHTML = await readFileAsync(path.join(outputDir, 'index.html'), { encoding: 'utf8' });
                expect(indexHTML).toMatchSnapshot();
            });

    }
);
