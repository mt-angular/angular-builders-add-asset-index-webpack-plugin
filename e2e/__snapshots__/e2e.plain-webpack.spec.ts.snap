// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = development End2End Snapshot index.html: AddAssetIndexPlugin default parameters with few blobs 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"assets/font/libre-franklin/libre-franklin-v2-latin-400.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-300.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-400.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-500.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-700.woff2\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
</body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = development End2End Snapshot index.html: AddAssetIndexPlugin default parameters with few blobs 2`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"assets/font/libre-franklin/libre-franklin-v2-latin-400.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-300.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-400.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-500.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-700.woff2\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
</body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = development End2End Snapshot index.html: AddAssetIndexPlugin default parameters with few files 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"assets/font/rubik/rubik-v7-latin-300.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-400.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-500.woff2\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
</body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = development End2End Snapshot index.html: AddAssetIndexPlugin default parameters with one blob 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"assets/font/libre-franklin/libre-franklin-v2-latin-400.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-300.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-400.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-500.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-700.woff2\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
</body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = development End2End Snapshot index.html: AddAssetIndexPlugin few blobs with asset specific options overriding generic options 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"public/deploy/libre-franklin/assets/font/libre-franklin/libre-franklin-v2-latin-400.8d503c823a91e889c0e7.woff2\\" integrity=\\"sha384-jVA8gjqR6InA55te1lTZNFqlUof0Y7M/qRbi0I+k1QzObyqva8Ue7Q0hOk2jrXMX\\" crossorigin=\\"anonymous\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
<link href=\\"public/deploy/rubik/assets/font/rubik/rubik-v7-latin-300.woff2\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"><link href=\\"public/deploy/rubik/assets/font/rubik/rubik-v7-latin-400.woff2\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"><link href=\\"public/deploy/rubik/assets/font/rubik/rubik-v7-latin-500.woff2\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"><link href=\\"public/deploy/rubik/assets/font/rubik/rubik-v7-latin-700.woff2\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"></body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = development End2End Snapshot index.html: AddAssetIndexPlugin one blob with generic options 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"public/deploy/assets/font/libre-franklin/libre-franklin-v2-latin-400.8d503c823a91e889c0e7.woff2\\" as=\\"font\\" rel=\\"preload\\"><link href=\\"public/deploy/assets/font/rubik/rubik-v7-latin-300.eb02199844b4e2a0871c.woff2\\" as=\\"font\\" rel=\\"preload\\"><link href=\\"public/deploy/assets/font/rubik/rubik-v7-latin-400.1ecc21093d3c8aa44b3e.woff2\\" as=\\"font\\" rel=\\"preload\\"><link href=\\"public/deploy/assets/font/rubik/rubik-v7-latin-500.d3fd715f234b8dcdd09d.woff2\\" as=\\"font\\" rel=\\"preload\\"><link href=\\"public/deploy/assets/font/rubik/rubik-v7-latin-700.9aff6f99df5790ad5dbd.woff2\\" as=\\"font\\" rel=\\"preload\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
</body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = production End2End Snapshot index.html: AddAssetIndexPlugin default parameters with few blobs 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"assets/font/libre-franklin/libre-franklin-v2-latin-400.8d503c823a91e889c0e7.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-300.eb02199844b4e2a0871c.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-400.1ecc21093d3c8aa44b3e.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-500.d3fd715f234b8dcdd09d.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-700.9aff6f99df5790ad5dbd.woff2\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
</body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = production End2End Snapshot index.html: AddAssetIndexPlugin default parameters with few files 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"assets/font/rubik/rubik-v7-latin-300.eb02199844b4e2a0871c.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-400.1ecc21093d3c8aa44b3e.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-500.d3fd715f234b8dcdd09d.woff2\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
</body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = production End2End Snapshot index.html: AddAssetIndexPlugin default parameters with one blob 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"assets/font/libre-franklin/libre-franklin-v2-latin-400.8d503c823a91e889c0e7.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-300.eb02199844b4e2a0871c.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-400.1ecc21093d3c8aa44b3e.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-500.d3fd715f234b8dcdd09d.woff2\\"><link href=\\"assets/font/rubik/rubik-v7-latin-700.9aff6f99df5790ad5dbd.woff2\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
</body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = production End2End Snapshot index.html: AddAssetIndexPlugin few blobs with asset specific options overriding generic options 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"public/deploy/libre-franklin/assets/font/libre-franklin/libre-franklin-v2-latin-400.8d503c823a91e889c0e7.woff2\\" integrity=\\"sha384-jVA8gjqR6InA55te1lTZNFqlUof0Y7M/qRbi0I+k1QzObyqva8Ue7Q0hOk2jrXMX\\" crossorigin=\\"anonymous\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
<link href=\\"public/deploy/rubik/assets/font/rubik/rubik-v7-latin-300.eb02199844b4e2a0871c.woff2\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"><link href=\\"public/deploy/rubik/assets/font/rubik/rubik-v7-latin-400.1ecc21093d3c8aa44b3e.woff2\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"><link href=\\"public/deploy/rubik/assets/font/rubik/rubik-v7-latin-500.d3fd715f234b8dcdd09d.woff2\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"><link href=\\"public/deploy/rubik/assets/font/rubik/rubik-v7-latin-700.9aff6f99df5790ad5dbd.woff2\\" as=\\"font\\" rel=\\"preload\\" network=\\"global\\" speed=\\"super-fast\\"></body>

</html>
"
`;

exports[`Test suite for e2e webpack config with AddAssetIndexPlugin. Webpack mode = production End2End Snapshot index.html: AddAssetIndexPlugin one blob with generic options 1`] = `
"<!doctype html>
<html lang=\\"fr\\">

<head>
    <meta charset=\\"utf-8\\">
    <title>IFA</title>
    <base href=\\"/\\">

    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=no\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">


<link href=\\"public/deploy/assets/font/libre-franklin/libre-franklin-v2-latin-400.8d503c823a91e889c0e7.woff2\\" as=\\"font\\" rel=\\"preload\\"><link href=\\"public/deploy/assets/font/rubik/rubik-v7-latin-300.eb02199844b4e2a0871c.woff2\\" as=\\"font\\" rel=\\"preload\\"><link href=\\"public/deploy/assets/font/rubik/rubik-v7-latin-400.1ecc21093d3c8aa44b3e.woff2\\" as=\\"font\\" rel=\\"preload\\"><link href=\\"public/deploy/assets/font/rubik/rubik-v7-latin-500.d3fd715f234b8dcdd09d.woff2\\" as=\\"font\\" rel=\\"preload\\"><link href=\\"public/deploy/assets/font/rubik/rubik-v7-latin-700.9aff6f99df5790ad5dbd.woff2\\" as=\\"font\\" rel=\\"preload\\"></head>

<body>
    <app-root class=\\"mt mat-typography\\"></app-root>
</body>

</html>
"
`;
